### PR TITLE
DAT-94 Improve error handling and add custom error for missing adxApatorId

### PR DESCRIPTION
**Pull request by rcrichton**
This allows for more sane and consistent error handling and returns a
descriptive error message when adx adapter ID is missing so that the
DATIM folks can manage and test the server more effectively.

### CODE REVIEW

File: src/server.coffee [hunk 1]
```diff
@@ -272,13 +280,13 @@ sendAsyncDhisImportResponseToReceiver = (out, res, callback) ->
...
 
-    callback true
+    callback()
 
 # post DXF data to DHIS2 api
 postToDhis = (out, dxfData, callback) ->
   if not dxfData
     out.info "No DXF body supplied"
-    return callback false
+    return callback new Error("No DXF body supplied")
```
Comment (BMartinos) [inline src/server.coffee]: Dont think we need to have the brackets here

File: src/server.coffee [hunk 2]
```diff
@@ -299,7 +307,7 @@ postToDhis = (out, dxfData, callback) ->
   request.post options, (err, res, body) ->
     if err
       out.error "Post to DHIS2 failed: #{err}"
-      return callback false
+      return callback new Error("Post to DHIS2 failed: #{err}")
```
Comment (BMartinos) [inline src/server.coffee]: Dont think we need to have the brackets here

**Comment by rcrichton** [comment]
@BMartinos cool thanks. Are you referring to where I do `callback()` above with null not explicitly passed? I think this is common practice if there are no other arguments to pass to the callback. The `err` comes through as undefined in this case and can still be tested for with `if (err)` correctly. I think this is common practice but happy to be proven otherwise.

**Comment by BMartinos** [comment]
Ok cool I'm fine with keeping it the way it is as its working as expected. Ive just recently been experiencing Lint errors when i have a empty callback with no error argument supplied. Happy to merge as is :+1:

**Pull request merged**